lib/ansible/module_utils/vmware_common/vmware.py

```diff
@@ -0,0 +1,1195 @@
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Joseph Callen <jcallen () csc.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import atexit
+import os
+import ssl
+import time
+
+try:
+    # requests is required for exception handling of the ConnectionError
+    import requests
+    from pyVim import connect
+    from pyVmomi import vim, vmodl
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+from ansible.module_utils._text import to_text
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.six import integer_types, iteritems, string_types
+
+
+class TaskError(Exception):
+    pass
+
+
+def wait_for_task(task):
+
+    while True:
+        if task.info.state == vim.TaskInfo.State.success:
+            return True, task.info.result
+        if task.info.state == vim.TaskInfo.State.error:
+            try:
+                raise TaskError(task.info.error)
+            except AttributeError:
+                raise TaskError("An unknown error has occurred")
+        if task.info.state == vim.TaskInfo.State.running:
+            time.sleep(15)
```

Comment:
How about doing a [truncated exponential backoff](https://cloud.google.com/storage/docs/exponential-backoff) here?

Reply:
@webknjaz This is a legacy code and I will address this in different PR.